### DISCUSSION TITLE
Data decentralisation

### DISCUSSION THREAD

**Discussion by PeterW-DASA** [m-ld/m-ld-spec#45] [Discussion]
If data is decentralised, eventually the different repositories will converge (assuming no changes are being made) - but how quickly does that happen? Presumably if it's not immediate, there is a risk that different users could end up viewing different versions of what they think is the same data, and perhaps making conflicting changes. How does the system help solve those problems?

**Comment by gsvarovsky** [comment]
Convergence happens as fast as the network can transmit a change, and the local persistence can ensure it's saved (the former likely to be dominant). Practically, this can often be taken as immediate.

However there's still the possibility of a 'conflict' if changes are sufficiently concurrent. In the most general case, m-ld simply surfaces the conflicting states to the app, which is responsible for dealing with them appropriately. However m-ld can also be configured to resolve conflicts automatically, with 'constraints'. This topic is discussed here: https://m-ld.org/doc/#concurrency (with particular reference to the "App Integrity" section).
